Update the Docker image versions that are pulled down

jsf9k opened the pull request:
## 🗣 Description ##

This pull request updates the versions of the Docker images that are pulled down to match those being used in [cisagov/guacamole-composition](https://github.com/cisagov/guacamole-composition).

## 💭 Motivation and context ##

If we don't pull the same versions used in [cisagov/guacamole-composition](https://github.com/cisagov/guacamole-composition) then the Docker composition won't work since the Guacamole instance does not have internet access.  It also adds a task to pre-create the `dbinit` directory with the correct permissions.

## 🧪 Testing ##

All automated testing passes.  I also built a staging AMI with these changes and verified that it functioned as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] _All_ future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] Tests have been added and/or modified to cover the changes in this PR.
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

- [x] Revert dependencies to default branches.

jsf9k committed:
Update the Docker images versions that are pulled down

These now agree with the versions used in
cisagov/guacamole-composition.

jsf9k committed:
Temporarily use a non-default branch of cisagov/guacamole-composition

Once the code in this branch has been successfully tested by building
a staging AMI and verifying that it functions as expected, this change
will be reverted.

jsf9k committed:
Add a task to pre-create the dbinit directory

This way we can set the permissions on the directory, instead of
letting it be created automatically with the default umask of 0022.

jsf9k committed:
Alphabetize Ansible task parameters on another file task

This way at least the two ansible.builtin.file tasks are consistent.

dav3r commented on tasks/main.yml:
```diff
@@ -34,9 +40,9 @@
+        - cisagov/guacscanner:1.1.11
+        - guacamole/guacd:1.4.0
+        - guacamole/guacamole:1.4.0
         # The version of the JDBC PostgreSQL driver included in the
```
Not related to this PR, but since it caught my eye, did you happen to check if the new Guacamole image contains an updated JDBC PostgreSQL driver?

https://github.com/cisagov/ansible-role-guacamole/blob/d6b9328139413cd5f109034320b96a0655445cdf/tasks/main.yml#L46-L50

jsf9k replied:
I did mean to check on that but forgot, so thanks for the reminder.  It looks like [version 1.4.0 of Guacamole is still using an ancient JDBC PostgreSQL driver](https://github.com/apache/guacamole-client/blob/1.4.0/guacamole-docker/bin/build-guacamole.sh#L111), but [the driver is updated to the most recent in the `master` branch](https://github.com/apache/guacamole-client/blob/master/guacamole-docker/bin/build-guacamole.sh#L111).

I have created cisagov/guacamole-composition#37 to track this work.  See also commit fa228ca.

dav3r replied:
Bummer- https://github.com/apache/guacamole-client/pull/655 with this change was merged into `master` on Dec. 4, 2021, but it wasn't included in [1.4.0](https://github.com/apache/guacamole-client/releases/tag/1.4.0) which was tagged on Jan. 2, 2022.

dav3r reviewed (approved):
🥑 👍

jsf9k committed:
Add an issue reference to a comment

This should remind us to upgrade to a newer PostgreSQL Docker image
the next time a new Guacamole Docker image is released.

mcdonnnj reviewed (approved):
Pinning intensifies.

jsf9k committed:
Revert "Temporarily use a non-default branch of cisagov/guacamole-composition"

This reverts commit 3f53b07e677eb7a4c630e6bf4f2d8d7f2268a8ba.
